### DISCUSSION TITLE
Rendering triangle in 3d with texture?

### DISCUSSION THREAD

**Discussion by HoutarouOreki** [raysan5/raylib#1612] [Q&A] [answered]
Hello. I have a simulated 3d world, in which ground is defined by triangles. I wanted to make a fast "3d preview" of it with Raylib, so naturally I would be using the DrawTriangle3D function to draw the ground, since it's made out of triangles. However I don't know how to make it have a texture. Is there a way to do it? Do I have to somehow convert the ground into a model and texture it? If it's possible, I would rather just slap a texture on a triangle.

**Comment by raysan5** [comment, marked as answer]
@HoutarouOreki raylib does not provide any direct function to map a texture into a triangle. To accomplish that you should use `rlgl` directly to define vertex position and texture coordinates assigned to every vertex. You can check [shapes.c->DrawTriangle()](https://github.com/raysan5/raylib/blob/master/src/shapes.c#L1195) for reference.

**Reply by KacperKotlewski** [reply]
> @HoutarouOreki raylib does not provide any direct function to map a texture into a triangle. To accomplish that you should use `rlgl` directly to define vertex position and texture coordinates assigned to every vertex. You can check [shapes.c->DrawTriangle()](https://github.com/raysan5/raylib/blob/master/src/shapes.c#L1195) for reference.
> 
> 

Hi! Did you know the solution with drawing each triangle from for example 10k triangles instead of makeing a single mesh make some preformence issues?

I think that should make preformence issue cuz of thousands of object in memory...

Second question if that make preformence issue there is a possibility to build in to raylib uv mapping to custom mesh models?

I want to make minecraft like system of generating world and in minecraft they have single mesh for each chunk, single texture boundle (the gigant texture png with block texture) and they map grass texture from texture map to exact coordinates on the mesh. 

I need high preformence system to make something thats similar to this and thats why I asking this two question :c otherwise i need to search other engine that's fits my needs.... I have trouble with running panda 3d in cpp and i search for alternative game frameworks
